### ISSUE TITLE
Compile Haxe to C++/Java/JS insted and then to native extension?

### ISSUE THREAD

**Issue by SKFrozenCloud** [hxdefold/hxdefold#38]
Haxe can be transpiled to C++, Java and Javascript.

Is it possible to do that and then convert it to a native extension instead?
The game performance and battery usage would be astonishing.

**Comment by haath** [comment]
Most likely this is possible, I don't see why it wouldn't be. 

But I think extensions are just extensions, you aren't *supposed* to script a game with them. They are mostly for writing specific systems and algorithms in a more optimized way. Also, every time you make a change to a native extension the Defold editor has to re-build the engine from the build server, and this would make development quite slow...

> The game performance and battery usage would be astonishing.

Even if you could script your entire game inside a native extension (and I'm not even sure that you can), are you sure that the performance improvements would be significant? I tend to think that with well-optimized scripts the performance would be roughly the same.
But to be sure you should probably try to find relevant information on the [defold forum](https://forum.defold.com/), or make a thread with your question.

---

Either way, this would require significant effort to provide the entire C++ API, and it would be a completely separate project to hxdefold. Personally I will not be working on this. Using Haxe instead of Lua is amazing because you get OOP features and type safety, but C++ already has that so I would just stick to writing C++ if need be 😅 

And as a side-note, while I don't have significant experience with Haxe->C++ yet, what I do remember is that working with pointers on the Haxe side gets a bit awkward. It would be much easier to just do it in C++.

Having said all that though, if somebody else wanted to work on this and makes a good proof of concept, we would happily include it in the hxdefold group 😄

**Comment by SKFrozenCloud** [comment]
Yeah true, I wanted to start a discussion but since there were no discussion tab in this repository, I made an issue.

**Comment by haath** [comment]
I think we can leave this open, as you said there could be a discussion about it 😉
